sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -598,6 +598,17 @@ object SQLConf {
       .bytesConf(ByteUnit.BYTE)
       .createOptional
 
+  val ADAPTIVE_MAX_SHUFFLE_HASH_JOIN_LOCAL_MAP_THRESHOLD =
+    buildConf("spark.sql.adaptive.maxShuffledHashJoinLocalMapThreshold")
+      .doc("Configures the maximum size in bytes per partition that can be allowed to build " +
+        s"local hash map. If this value is bigger than ${ADVISORY_PARTITION_SIZE_IN_BYTES.key} " +
+        s"and all the partition size not larger than ${ADVISORY_PARTITION_SIZE_IN_BYTES.key}, " +
+        "join selection prefer to use shuffled hash join instead of sort merge join regardless " +
+        s"of the value of ${PREFER_SORTMERGEJOIN.key}.")
+      .version("3.2.0")
+      .bytesConf(ByteUnit.BYTE)
+      .createOptional
```

Comment:
shall this fallback to `ADVISORY_PARTITION_SIZE_IN_BYTES`? `.fallbackConf(ADVISORY_PARTITION_SIZE_IN_BYTES)`

Reply:
If we do `fallbackConf`, many tests will fail since smj changes to shj. Not sure we should enable this feature by default.

Reply:
I think we should, but let's do it in a followup PR with more discussion.

Reply:
then shall we use 0 as the default value to turn it off?

Reply:
I'm ok with it